Update non-Rust portion of codebase for Godot 4

skison opened the pull request:
- Fixed & refactored all project examples (some subjective implementation changes too)
- Updated GUT dependency & unit tests
- Updated C# wrapper & visualization example
- Reintroduced addons directory with GDExtension requirements
- Updated Readme details & formatting/wording
- Updated file & directory names to better match Godot 4 style guide
- Updated gitignore to include asset imports and discard some unnecessary file types

skison committed:
Fix and refactor both example project implementations

skison committed:
Update all visualization code for Godot 4, re-add C# support

skison committed:
C# file fixes

skison committed:
Update Unit testing (GUT) for Godot 4

Some minor fixes are still needed for the DijkstraMap API for the unit tests to succeed: add get_terrain_for_point method (and un-comment that section in unit test file), and fix default terrain id -1 not having a default weight of 1.0 instead of INF (it is the only one that behaves like this)

skison committed:
Update & format Readme

skison committed:
Update gitignore and include .import files

skison committed:
Format GDScript and document gdformat usage

skison committed:
Fix visualization_demo path

skison committed:
Update reference to GDExtension in Readme

skison committed:
Fix filename case sensitivity,  gitattribute exclusions, and Readme wording

skison committed:
Delete redundant Tests directory

(this was meant to be renamed to 'tests' but git considered it a separate directory)

skison committed:
Add gut config to gitattributes export ignore

astrale-sharp commented:
Hey there, thanks for your amazing work so far! :D

i tested your PR and I have a few notes

tests are supposed to be automatically generated, so i wouldn't bother fixing them if I were you (still an amount of them that doesn't pass)

all the examples are working, good job!

I have not tested C# as I don't have the right version of the engine!
it's also not a blocker, you don't have to fix everything at once!

The right repo to open your PR (the main one) is https://github.com/MatejSloboda/Dijkstra_map_for_Godot , which I can merge anyway

skison commented:
Thanks for reviewing it!

I don't know if it makes sense to merge to my changes to the main repo yet since this is all built on the Godot 4 updated version of DijkstraMap which right now only lives in your fork/branch; wouldn't we want to either merge that to the main repo first (probably not ideal since there are still some minor things to fix on the Rust side), or merge the changes here first?

astrale-sharp commented:
We can merge to the main repo as long as we don't release it ! also, if it's merged here then I merge it from here there, you won't appear in the main repo's stats which is not fair ^^

Odebe committed:
fix the exact gdnext version

astrale-sharp commented:
Hey again there, maybe open the pull request on the main repo as I just force pushed on this one ;)

skison commented:
I'm glad to see there is some new activity! 😁

If we're working toward a Godot 4.3 (or 4.4) compatible release now, I think I'll need to revisit the code and make some adjustments - the demo scenes may be broken now since they changed how TileMaps work in 4.3 (but it shouldn't be too hard to fix). So maybe we should decide on a specific version of both Godot and gdext to target?

Once we figure out what to do then yeah, I'll open the real PR! Maybe in it we can include a checklist of to-do items before release, or make that separately.

astrale-sharp commented:
Same, my main blocker is that my real computer is broken for 2.5 more weeks :/
programming on the secondary one is barely possible

skison committed:
Update for Godot 4.4.1

- Update the codebase to Godot 4.4.1
- Pin gdext-rust version to 0.2.4
- Minor fixes to rust & gdscript code
- Add tool script (tools/build_platform.sh) to simplify generation of debug/release builds for windows, linux, and mac (requires the host platform to support these options)
- Generate updated debug & release builds for Windows

skison committed:
Add support + builds for Linux & WebAssembly

Update the build script to support experimental WebAssembly debug & release builds
Export debug & release binaries for Linux, WebAssembly unthreaded, and WebAssembly threaded platforms
Minor rework of visualization demo to avoid crashes on non-C# supported platforms
README updates

skison commented:
Hey, I pushed some pretty big updates on my end - everything is now up-to-date with Godot 4.4.1, and there are new debug & release builds for Windows, Linux, and even WebAssembly! The web builds aren't very efficient but they do work, both threaded and unthreaded versions. Unfortunately I don't have a way to build for macOS, so those binaries are missing for now. I created a new build tool script at `tools/build_platform.sh` that makes the whole build process easier and hopefully will make it easier to test Rust code changes going forward!

Grublady committed:
Update actions/upload-artifact to v4 (#2)

skison committed:
Update project + GUT addon to 4.5.1

skison committed:
Update gdext to v0.4.0 + GitHub Actions updates (#3)

- gdext version now pinned to v0.4.0
	- Minor Rust code fix to support this version (replaced deprecated `godot_name()` method)
- Updated + fixed build action
  - Added both debug + release build commands for every platform
  - Added MacOS ARM platform builds
  - Added placeholder web threads + nothreads builds
 	 - Currently disabled due to failed build attempts; see [this bug report](https://github.com/godot-rust/gdext/issues/1356). For now we can fallback to the previous test web binaries.
  - Upgraded download-artifact@v3 to v4
  - Add steps to rename binaries to standardized format
  - Improve Windows builds via windows build platform + msvc target
- Update manual binary check action to consider all platform binaries

skison committed:
Update to gdext v0.5.3 + support optional params

skison committed:
Update GUT to 9.6.0 (for Godot 4.6)

skison committed:
API + test fixes

skison committed:
Use gdext register-docs feature + minor API/Godot docs fixes

skison committed:
Update build scripts w/web support + asset reimports

skison committed:
Release Godot 4.6 debug
Test binaries for all platforms for Godot 4.6 + gdext v0.5.3

skison committed:
AnyDictionary type fix + enforce GDScript static typing

skison committed:
Examples fixes + GDScript format

skison committed:
Release Godot 4.6 debug 2
Test binaries for all platforms for Godot 4.6 + gdext v0.5.3 (minor fixes)

skison committed:
Minor examples cleanup
